src/main/java/racingcar/RacingCarGameLogic.java

```diff
@@ -0,0 +1,31 @@
+package racingcar;
+
+import racingcar.model.Car;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class RacingCarGameLogic {
+    private static RacingCarGameLogic instance = new RacingCarGameLogic();
+
+    private RacingCarGameLogic() {
+
+    }
+
+    public static RacingCarGameLogic getInstance() {
+        return instance;
+    }
+
+    public void gameLogic(int numberOfCar, int numberOfAttempt) {
+        List<Car> cars = initCars(numberOfCar);
+        IntStream.range(0, numberOfAttempt).forEach(i -> RacingCarGameRule.getInstance().moveCars(cars));
+    }
+
+    private List<Car> initCars(int numberOfCar) {
+        List<Car> cars = new ArrayList<>(numberOfCar);
+        IntStream.range(0, numberOfCar).forEach(i -> cars.add(new Car()));
+        return cars;
+    }
```

Comment:
Stream을 사용하셨네요 🫢
작성하신 코드와 for문으로 돌리는 방법간 차이가 없어보입니다. 오히려 for문을 사용했더라면 더 읽기 편했을 것 같아요

컬렉션을 만들 때 forEach는 거의 사용하지 않습니다.
스트림을 만들고 리스트를 만드는 방법이 오히려 가독성 측면에서 우월합니다.

```suggestion
    private List<Car> initCars(int numberOfCar) {
        return Stream.generate(Car::new)
                .limit(numberOfCar)
                .collect(Collectors.toList());
    }
```